CLI: Adjust PRO flags as requested by Product

IagoAbal opened the pull request:
Closes PA-2488

test plan
---------
% semgrep --pro-languages -l apex -e 'foo(...)' \
          semgrep-proprietary/tests/patterns/apex/dots_string.cls
^ finds stuff
% semgrep -c p/default-v2 \
      examples-annotations/repos/java/SQLi/vuln-repos/WebGoat
^ 44 findings, ~100s
% semgrep -c p/default-v2 --pro-intrafile \
      examples-annotations/repos/java/SQLi/vuln-repos/WebGoat
^ 60 findings, ~100s
% semgrep -c p/default-v2 --pro \
      examples-annotations/repos/java/SQLi/vuln-repos/WebGoat
^ 60 findings, ~500s

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

IagoAbal commented on cli/src/semgrep/constants.py:
```diff
@@ -32,13 +32,13 @@
-    INTERFILE = auto()
+    PRO_LANG = auto()
+    PRO_INTRAFILE = auto()
+    PRO_DEEP = auto()
```
Or `PRO_INTERFILE` ? Somebody had observed that "intrafile" and "interfile" were very similar. 🤷‍♂️ Another option could be `PRO_FULL` since the idea is that it provides the "full experience".

nmote commented on cli/src/semgrep/commands/scan.py:
```diff
@@ -606,19 +606,31 @@ def scan_options(func: Callable) -> Callable:
+    # help="contact support@r2c.dev for more information on this"
+)
+@click.option(
+    "--pro-intrafile",
```
What is the difference between this and `--interproc`?

IagoAbal replied:
No difference, just synonyms. I'm a bit confused by so many flag renaming, I'm not sure whether to keep it around or deprecate `--interproc` and `--interfile` like we did with `--deep`.

nmote replied:
We should have exactly one non-deprecated flag for each behavior. We can keep the old ones around for a while, but we should clearly mark them as deprecated in the help and point people towards the preferred flag.

IagoAbal replied:
OK added help messages for the three `--pro*` ones and deprecated the others.

nmote reviewed (commented):
Get tests green and I'll take another look

nmote reviewed (changes requested):
Tests are still red

IagoAbal committed:
Nat's review

github-actions[bot] committed:
Update pytest snapshots

IagoAbal committed:
CLI: Adjust PRO flags as requested by Product

Closes PA-2488

test plan
---------
% semgrep --pro-languages -l apex -e 'foo(...)' \
          semgrep-proprietary/tests/patterns/apex/dots_string.cls
^ finds stuff
% semgrep -c p/default-v2 \
      examples-annotations/repos/java/SQLi/vuln-repos/WebGoat
^ 44 findings, ~100s
% semgrep -c p/default-v2 --pro-intrafile \
      examples-annotations/repos/java/SQLi/vuln-repos/WebGoat
^ 60 findings, ~100s
% semgrep -c p/default-v2 --pro \
      examples-annotations/repos/java/SQLi/vuln-repos/WebGoat
^ 60 findings, ~500s

nmote reviewed (approved):
Looks good, thanks!
